src/components/SocialListItem.tsx

```diff
@@ -1,42 +1,64 @@
 // Libraries
-import React from "react"
-import styled from "@emotion/styled"
+import React, { useMemo } from "react"
+import { Flex, Box, chakra, Icon } from "@chakra-ui/react"
+import {
+  FaTwitter,
+  FaYoutube,
+  FaDiscord,
+  FaRedditAlien,
+  FaStackExchange,
+  FaGlobe,
+} from "react-icons/fa"
 
 // Components
-import Icon from "./Icon"
+import { socialColors } from "./Icon"
 
 // Styles
-const Item = styled.div`
-  display: flex;
-  width: 100%;
-  padding: 0.5rem 0;
-  align-items: center;
-`
-
-const StyledIcon = styled(Icon)`
-  padding-right: 0.75rem;
-`
-
-const ChildrenContainer = styled.div`
-  font-style: italic;
-
-  > a {
-    font-style: normal;
-  }
-`
+const StyledIcon = chakra(Icon, {
+  pr: 3,
+  boxSize: "2.5rem",
+})
 
 export interface IProps {
   children?: React.ReactNode
   socialIcon: string
 }
+
 const SocialListItem: React.FC<IProps> = ({ children, socialIcon }) => {
+  const RenderIcon = useMemo(() => {
+    switch (socialIcon) {
+      case "reddit":
+        return (
+          <StyledIcon as={FaRedditAlien} color={socialColors[socialIcon]} />
+        )
+      case "twitter":
+        return <StyledIcon as={FaTwitter} color={socialColors[socialIcon]} />
+      case "youtube":
+        return <StyledIcon as={FaYoutube} color={socialColors[socialIcon]} />
+      case "discord":
+        return <StyledIcon as={FaDiscord} color={socialColors[socialIcon]} />
+      case "stackExchange":
+        return (
+          <StyledIcon as={FaStackExchange} color={socialColors[socialIcon]} />
+        )
+      case "webpage":
+        return <StyledIcon as={FaGlobe} color={socialColors[socialIcon]} />
+    }
+  }, [socialIcon])
```

Comment:
We should have this mapping outside the component. No need to have it in a memo.

```tsx
const icons = {
  reddit: FaRedditAlien,
  ...
}

const socialColors = {
  reddit: '#...',
  ...
}

// then inside the component
const SocialIcon = icons[socialIcon]
const color = socialColors[socialIcon]
<Icon as={SocialIcon} color={color}>
```